### DISCUSSION TITLE
CSP build on NPM

### DISCUSSION THREAD

**Discussion by chadanuk** [alpinejs/alpine#1944] [7. Bugs (archived)]
### Discussed in https://github.com/alpinejs/alpine/discussions/1929

<div type='discussions-op-text'>

<sup>Originally posted by **tomb1n0** August 10, 2021</sup>
First of all, thanks for alpine - i'm absolutely loving using it in combination with blade components and inside livewire!

As far as I can see the CSP version of alpine isn't up on NPM, howeve plugins like intersect and persist are.

Is there any reason for this or is it expected that developers should build the CSP version themselves? if this is the case the documentation doesn't make this clear :(</div>

**Comment by haarts** [comment]
In the meanwhile how DO you build the CSP version yourself?

**Comment by mhio** [comment]
So this was a discussion, turned into an issue, that was closed and moved into a new discussion??

In any case, I ran into the same issue:

```
git clone https://github.com/alpinejs/alpine
cd alpine
npm install
npm run build
```

```
$ find packages/csp/dist -ls
   524806      4 drwxr--r--   2 root     root         4096 Nov 29 06:18 packages/csp/dist/
   524854     36 -rw-r--r--   1 root     root        36216 Nov 29 06:18 packages/csp/dist/cdn.min.js
   524859     88 -rw-r--r--   1 root     root        88377 Nov 29 06:18 packages/csp/dist/cdn.js
   524856    104 -rw-r--r--   1 root     root       105845 Nov 29 06:18 packages/csp/dist/module.esm.js
   524855    104 -rw-r--r--   1 root     root       106164 Nov 29 06:18 packages/csp/dist/module.cjs.js
```

**Reply by mhio** [reply]
I should add, you probably want to checkout a release tag before building
```
git checkout v3.5.2
```

**Comment by tomb1n0** [comment]
Yeah i feel like there has to be something missing here. If it's expected for developers to build this themselves that's fine but it'd be nice if the documentation made this clear..

**Comment by thibaudcolas** [comment]
I got really confused by this as well so have made a PR updating the docs (https://github.com/alpinejs/alpine/pull/2619). FYI there are also a lot of people asking for this in https://github.com/alpinejs/alpine/issues/237.

**Comment by thibaudcolas** [comment]
I made a build for myself and thought I’d share in case it helps others. It’s from commit d49a0b0b5841d05e41b70dc806d80c2208765666, here is a ZIP:

[alpinejs-csp-3.0.0-alpha.0.tgz.zip](https://github.com/alpinejs/alpine/files/7954738/alpinejs-csp-3.0.0-alpha.0.tgz.zip)

<details><summary>Commands I ran to get this</summary>

```bash
git clone git@github.com:alpinejs/alpine.git
cd alpine
nvm use 16
npm install
npm run build
cd packages/csp
npm pack --loglevel notice 2>&1 >/dev/null | sed -e 's/^npm notice //'
```

</details>

<details><summary>Logs of `npm pack`</summary>


```
📦  @alpinejs/csp@3.0.0-alpha.0
=== Tarball Contents ===
106B    builds/cdn.js
59B     builds/module.js
92.3kB  dist/cdn.js
38.1kB  dist/cdn.min.js
109.8kB dist/module.cjs.js
109.5kB dist/module.esm.js
306B    package.json
1.3kB   src/index.js
=== Tarball Details ===
name:          @alpinejs/csp
version:       3.0.0-alpha.0
filename:      @alpinejs/csp-3.0.0-alpha.0.tgz
package size:  87.7 kB
unpacked size: 351.4 kB
shasum:        8f5f42b27d88eb2f3c96042dedb83c5aa67253e2
integrity:     sha512-U9FlqTxU+OXJK[...]9uYcFNDFMAaGg==
total files:   8

alpinejs-csp-3.0.0-alpha.0.tgz
```

</details>

I’ve also published it to npm as [@thibaudcolas/alpinejs-csp-build](https://www.npmjs.com/package/@thibaudcolas/alpinejs-csp-build). Here is a CodePen with it up and running: https://codepen.io/thibaudcolas/pen/qBVdoXp.

**Reply by SimoTod** [reply]
It's worth pointing out that x-model doesn't work in the csp build at the moment so other stuff may be incomplete as well.

**Comment by thibaudcolas** [comment]
Adding links to all the relevant threads on this for future reference:

- https://github.com/alpinejs/alpine/discussions/3011
- https://github.com/alpinejs/alpine/discussions/2773
- https://github.com/alpinejs/alpine/discussions/2724
- https://github.com/alpinejs/alpine/discussions/1867
- https://github.com/alpinejs/alpine/discussions/2395
- https://github.com/alpinejs/alpine/discussions/2650
- https://github.com/alpinejs/alpine/discussions/1529

**Comment by ecmanriquejrdev** [comment]
Is there any updates regarding the npm Content Security Policy (CSP) related issues of alpinejs

**Reply by thibaudcolas** [reply]
For future ref – the CSP build was made as of #3959 in January 2024, and is available for all versions since v3.13.3 🌈
